Added SortedMap

eldario opened the pull request:
Added implementation of SortedMap

TODO:
- [ ] tests
- [ ] benchs
- [x] makefile

eldario committed:
Added SortedMap

eldario committed:
Added reading from file

eldario committed:
Removed sample file

eldario committed:
Move to new method

korjavin reviewed (changes requested):
Мне это решение показалось сложным в плане цикломатической сложности.

Давай его побенчмаркаем прежде чем изменять.

eldario committed:
Added 2 benchmarks
Added result.md file with result output
Configure go.mod
Adde Makefile
Move logic from core to packet

eldario commented:
@korjavin 
#### Benchmarks

```json
BenchmarkInsert-4                4746104               262 ns/op
BenchmarkGetFrequentUses-4      12489802               100 ns/op
```

eldario committed:
Added new packet with SimpleImplementation
Move reader from core to packet
Added new dependency for styling output
Added flags from command
Add some benchs
Added linter

eldario committed:
Added Remove method to Mapper packet

korjavin commented on task3/pkg/reader/reader.go:
```diff
@@ -0,0 +1,94 @@
+		return false
+	}
+
+	if _, err := strconv.Atoi(word); err == nil {
```
Это мне кажется тяжелой процелурой

korjavin commented on task3/pkg/simpleMapper/simpleMapper.go:
```diff
@@ -0,0 +1,94 @@
+ * Structure constructor.
+ */
+func New() *SortedMap {
+	return &SortedMap{words: make(map[string]int), topCount: 8}
```
8 - подозрительно выглядит

eldario replied:
Debug mode был, убрал дефолт

korjavin reviewed (commented):
я ещё посмотрю

eldario committed:
Removed default value from Mapper

eldario committed:
Modified exclude list

eldario committed:
Update with filtering

eldario committed:
Update results table

eldario commented:
@korjavin 

```json
ID  Word    Count  Order  
1   like    35     56     
2   told    29     76     
3   looked  39     90     
4   marry   21     133    
5   went    62     139    
6   love    32     175    
7   want    29     182    
8   into    29     188    
9   took    20     218    
10  cant    18     335    
```

eldario committed:
Try to remove own dependency from pkg
Improve collect of word from text

eldario committed:
Refactor

eldario committed:
Fix with word validation

eldario committed:
Remove redaunt condition with check the last symbol
